src/main/java/controller/ListMessages.java

```diff
@@ -199,15 +204,31 @@ public void listMessagesWithoutChatIdPathVariable(HttpServletRequest request) {
          * If it is null, sets it to the time when listMessages was called.
          */
         List<Message> messageListForReceivedTsUpdate = new ArrayList<Message>();
+        List<Long> attachmentIdList = new ArrayList<Long>();
 
         for (Message message : messages) {
             if (message.getReceivedTs() == null && message.getSenderId() != userId) {
                 message.setReceivedTs(receivedTs);
                 messageListForReceivedTsUpdate.add(message);
             }
+
+            if (message.getAttachmentId().isPresent()) {
+                attachmentIdList.add(message.getAttachmentId().get());
+            } 
         }
         
         insertMessage.updateReceivedTsForMessages(messageListForReceivedTsUpdate);
+
+        if (!attachmentIdList.isEmpty()) {
+            List<Attachment> attachments = queryAttachment.getAttachments(attachmentIdList);
+            Map<Long, Integer> attachmentIdToIndexInList = new HashMap<Long, Integer>();
+
+            for (int i = 0; i < attachments.size(); ++i) {
+                attachmentIdToIndexInList.put(attachments.get(i).getAttachmentId(), i);
```

Comment:
instead of index in the list, shouldn't we use the message id instead?
that is a more reliable and stable id than index.

Reply:
We would be using this map `attachmentIdToIndexInList` in `SuccessResponseGenerator`. For every message, we already have the corresponding `attachmentId` in the `Message` object itself; we need the index of the `Attachment` object with that `attachmentId`, so that we can add it to the response without having to search though the list `attachments`. 

If we don't want to use the `index` in the list, an alternative way could be `attachmentId` as the `key` and all `Attachment` details as its `value` in the `Map`. Should I make this change?

Reply:
yes, we should.
index is not a stable and reliable id.
`attachmentId` is better and more reliable.

Reply:
Done.